### PR TITLE
User provided certs

### PR DESCRIPTION
**Pull request by fgiorgetti**
Allow users to provide their own certificates to be used with non-kube sites in Skupper V2.
More information is available through: cmd/bootstrap/PROVIDED_CERTIFICATES.md

### CODE REVIEW

File: cmd/bootstrap/PROVIDED_CERTIFICATES.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+for example, at the time a local site is being initialized, a bundle is being prepared
+or a bundle is being installed at a remote location.
+
+To understand it better, let's go through the main use cases and review what is the ideal
+phase that a given kind of certificate should be provided.
+
+### Using a custom CA to sign certificates
+
+If you want Skupper to generate client and server certificates signed by a custom CA,
+you will need to provide the respective certificates during:
```
Comment (grs) [inline cmd/bootstrap/PROVIDED_CERTIFICATES.md:40]: I presume the custom CA needs to use a specific filename? Or are they provided in the form of a kubernetes secret with a given object name?

File: cmd/bootstrap/PROVIDED_CERTIFICATES.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+### Using custom server and client certificates
+
+If you want a new local site to use your custom server and client certificates, you can
+provide them at any time, for example:
+
+* Local site initialization time
+* Site bundle preparation time
+* Site bundle installation time
+
+During "Local site initialization time" or "Site bundle preparation time", Skupper will detect
```
Comment (grs) [inline cmd/bootstrap/PROVIDED_CERTIFICATES.md:60]: Again, maybe worth clarifying here *how* it detects this? Also if it detects a server certificate will the CA still be generated or is that then deemed unnecessary?

File: cmd/bootstrap/PROVIDED_CERTIFICATES.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+## Provide your own skupper-site-ca
+
+If you want Skupper to use your own CA certificates to generate and sign server and client
+certificates used for site linking, you can simply create the following structure under
+the namespace home of your choice, for example:
+
+```shell
+${HOME}/.local/share/skupper/namespaces/default/input/certificates/
+└── ca
+    └── skupper-site-ca
```
Comment (grs) [inline cmd/bootstrap/PROVIDED_CERTIFICATES.md:86]: This mostly answers my question above, but perhaps worth clarifying that the name of the directory is important here?

**Review by grs — APPROVED** [review]

File: cmd/bootstrap/PROVIDED_CERTIFICATES.md (hunk 2 above)
Comment (fgiorgetti) [reply cmd/bootstrap/PROVIDED_CERTIFICATES.md:60]: I will update it, explaining how they are detected.

Just answering your question, the CA is always generated (if not provided).

The server certificate is currently associated with a RouterAccess, so users might eventually
provide server and client certificates to a given RouterAccess but they could have a different
RouterAccess defined that uses certificates generated by the default CA.

File: cmd/bootstrap/PROVIDED_CERTIFICATES.md (hunk 3 above)
Comment (fgiorgetti) [reply cmd/bootstrap/PROVIDED_CERTIFICATES.md:86]: Agreed. The directory name is actually what identifies the certificate.
I will add that information earlier in the document to make it clear.

**Review by grs — APPROVED** [review]

**Review by nluaces — APPROVED** [review]

**Review by ajssmith — APPROVED** [review]

**Review by c-kruse — APPROVED** [review]
Tested with user provided site CA!

**Review by lynnemorrison — APPROVED** [review]

**Pull request merged**